runtime/lua/vim/lsp/util.lua

```diff
@@ -896,11 +896,12 @@ end
 ---@param offset_encoding string utf-8|utf-16|utf-32 (required)
 --see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_applyEdit
 function M.apply_workspace_edit(workspace_edit, offset_encoding)
-  if offset_encoding == nil then
+  if not offset_encoding then
     vim.notify_once(
       'apply_workspace_edit must be called with valid offset encoding',
-      vim.log.levels.WARN
+      vim.log.levels.ERROR
     )
+    return
```

Comment:
`apply_workspace_edit` most use in a callback  if there not pass a offset_encoding this function should not run right ? current buffer may not same as before. so i think can't set the first client `offset_encoding` in this function .  any thought?

Reply:
It will currently fallback to `utf-16`, which is correct for almost all LSPs (clangd or rust-analyzer _can_ run with utf-8, but the neovim defaults to utf-16). I think turning this into an error with an early exit makes things worse.
We shouldn't do that before we haven't fixed all usages

(And it's general not nice to mix behavior changes into a generic cleanup PR focused on type annotation fixes)